lib/chef/knife/windows_cert_generate.rb

```diff
@@ -75,7 +76,7 @@ def prompt_for_passphrase
       end
 
       def generate_certificate rsa_key
-        @hostname = "*"
+        @hostname = Socket.gethostname
```

Comment:
It's definitely your call @btm -- we can always add a default later if we decide that's helpful. My last feedback on this is that on Windows systems, DNS is almost always Active Directory, and particularly for domain-joined systems, the fqdn can be obtained very reliably. Also, Windows users often *don't* really know what the fqdn is, particularly in domains that are actually subdomains (they don't always know they're in a subdomain) and not the root of a forest.

Regardless, I think we can give good instructions here and if we find data indicating this is confusing (e.g. mailing list questions, etc.), we can revisit -- adding a default wouldn't break compatibility.

Thanks for looking at this!